crates/op-rbuilder/src/monitor_tx_pool.rs

```diff
@@ -0,0 +1,43 @@
+use futures_util::StreamExt;
+use reth_optimism_node::txpool::OpPooledTransaction;
+use reth_transaction_pool::{AllTransactionsEvents, FullTransactionEvent};
+use tracing::debug;
+
+pub async fn monitor_tx_pool(mut new_transactions: AllTransactionsEvents<OpPooledTransaction>) {
+    while let Some(event) = new_transactions.next().await {
+        transaction_event_log(event);
+    }
+}
+
+fn transaction_event_log(event: FullTransactionEvent<OpPooledTransaction>) {
+    match event {
+        FullTransactionEvent::Pending(hash) => {
+            debug!("Transaction event: tx={:?}, kind=pending", hash)
```

Comment:
this wouldn't work with json structured logging?
```suggestion
            debug!(
                    tx_hash=tx_hash.to_string(),
                    kind = "pending",
                    "Transaction event received"
                )
```

Reply:
I will change this once I am back at home